### PR TITLE
fix(web): self-heal zombie structured-view WebSocket behind a proxy

### LINKED ISSUE #2287: Structured view: live updates silently stall after WS idle-reset behind a proxy; only resync on user action (apparent regression of #1130)
**Note:** Investigated with Claude Code assistance against a live 1.11.1 deployment; decisions and report are mine.

Looks like an apparent regression / incomplete coverage of #1130 (which fixed `useCockpit` reconnect for the old cockpit). On the **1.11.1 structured view**, the live WebSocket still dies on idle and the UI does not auto-reconnect — with a subtler symptom than #1130.

## Symptom
On a remote (behind-proxy) deployment, returning to a session tab shows **no new messages** even though the agent has been producing output. There's **no obvious "disconnected" indicator** — the view just silently stops updating. **Sending a message (or any action) triggers a reconnect**, and all buffered events flush in at once. No data is lost (events are persisted and replayed); the live view is simply stale until the user acts.

## Environment
- aoe `1.11.1` (`1.11.1+g9d37e6bee953`), Linux host
- `aoe serve --auth=passphrase --behind-proxy`, fronted by **Tailscale Serve** (HTTPS tailnet → `http://127.0.0.1:8080`)
- Clients: Chrome/Edge 149 (Windows desktop) and Android Chrome (PWA)

## Repro
1. Run `aoe serve --behind-proxy` behind a reverse proxy with a WS idle timeout (Tailscale Serve, Cloudflare, etc.).
2. Open a session in the structured view; leave the tab idle/backgrounded while the agent keeps emitting output.
3. Return to the tab → no new messages.
4. Send a message → view reconnects and all missed messages appear together.

## Evidence
Daemon log shows the browser WS being reset abruptly, repeatedly (33× over ~3 days in one `debug.log`):

```
WARN acp.ws: client recv error: WebSocket protocol error: Connection reset without closing handshake
```

"Reset without closing handshake" = the proxy/network dropped the idle connection (RST), not a clean close — consistent with proxy idle-timeout and/or background-tab socket suspension, the same mechanism as #1130 (there: Cloudflare 100s idle + mobile background; here: Tailscale Serve + desktop/mobile idle).

## Suspected cause
1. The structured-view WS client has no reconnect path on `close`/`error`, and no `visibilitychange` / `online` listener — so it only re-opens when the user switches sessions or hard-refreshes (the same gap #1130 fixed for `useCockpit`, now present on the structured-view hook).
2. The daemon doesn't appear to send periodic application-level WS **Ping** frames to browser clients, so idle connections are dropped by intermediaries. No server-side keepalive/ping knob is exposed (`aoe serve --help` / `config.toml`).

## Suggested fix
- **Frontend:** give the structured-view WS hook the same treatment as the #1130 `useCockpit` fix — reconnect-with-backoff on `close`/`error`, plus `document.visibilitychange` (refocus) and `window.online` listeners, with a visible "reconnecting" state.
- **Daemon:** emit periodic WS Ping frames (~20–30s) to client connections to survive proxy idle timeouts; optionally expose the interval via config.

## Impact
Affects all remote/behind-proxy users. Worse than #1130 in one respect: there's no clear disconnected indicator, so users don't realize the view is stale.

### PR DESCRIPTION
**Pull request by Seluj78**
**Note:** Claude handled the implementation and prose via back-and-forth. Idea, decisions, and everything else are mine. — @Seluj78

## Description

On a remote, behind-proxy deployment (Tailscale Serve, Cloudflare), the structured-view live WebSocket silently stalls: you return to a session tab and see no new messages, with no disconnected indicator. Sending a message reconnects and flushes everything at once. No data is lost (events are persisted and replayed), the live view is just stale until you act.

Root cause is a half-open (zombie) socket, not the gaps the report first suspected. The #1130 client reconnect (visibilitychange / online / backoff) and the daemon WS Ping keepalive both already shipped in 1.11.1. They do not help here because: the proxy RSTs the idle leg, the daemon's Close send fails silently on the dead socket, and the browser never receives `onclose`, so the WS sits in `readyState === OPEN` forever. Browser JavaScript cannot observe WS Ping/Pong frames, so a quiet session has no liveness signal, and the visibility/online reconnect bailed whenever `readyState === OPEN`. Nothing re-dialed until a user send finally surfaced the dead socket.

This PR makes the view self-heal:

- **Daemon:** emit a small `{"kind":"heartbeat"}` Text frame on the existing 30s ping tick, alongside the WS Ping. The Ping stays for the server-side pong reaper; the Text frame is the JS-observable liveness signal.
- **Web:** track the timestamp of the last message received on the current socket (bumped on every frame, including the heartbeat). A 15s watchdog and the visibility/online triggers re-dial when an `OPEN` socket goes quiet past 75s (two missed heartbeats plus slack, under the daemon's 90s reap). The watchdog is gated on `readyState === OPEN`, so it heals zombies but never resurrects an intentionally-closed or retry-exhausted socket. Re-dial routes through `connect()`, which closes the zombie and bumps the dial generation, so recovery never depends on the dead socket firing `onclose`. `connect()` now bumps that generation before closing the prior socket so a synchronously-firing `onclose` is correctly orphaned.

A false-positive reconnect is cheap: the re-dial resumes from `?since=<lastSeq>` and dedupes.

Closes #2287

## PR Type

- [ ] New Feature
- [x] Bug Fix
- [ ] Refactor
- [ ] Documentation
- [ ] Infrastructure / CI

## How to test

- [ ] Run `aoe serve --behind-proxy` behind a reverse proxy with a WS idle timeout (Tailscale Serve, Cloudflare).
- [ ] Open a session in the structured view and leave the tab foregrounded but idle while the agent keeps emitting output. Within ~75s of the socket dying, the view re-dials on its own and missed messages flush, without sending anything.
- [ ] Background the tab, let the socket die, then refocus: live updates resume immediately (no message send needed).
- [ ] Leave a quiet, healthy session open for several minutes: it stays connected and does not spuriously flicker into "Reconnecting".
- [ ] Confirm the daemon `debug.log` still shows the keepalive working and the view recovers after a `Connection reset without closing handshake`.

## Checklist

- [x] New and existing tests pass
- [x] Documentation was updated where necessary
- [ ] For UI changes: included screenshot or recording

## Test Coverage Analysis

**Web dashboard / structured view (Playwright user story)**
- [x] Added or updated a test and updated `web/tests/coverage-matrix.json`

Covered with Vitest rather than Playwright: a half-open zombie socket (browser still reports `OPEN` after a silent proxy RST) cannot be produced through a real browser in Playwright, but the `FakeWebSocket` harness in `web/src/hooks/useAcpSession.reconnect.test.ts` drives it exactly. New cases: (a) a stale `OPEN` socket with no heartbeat re-dials on its own (verified red on the pre-fix tree), (b) heartbeats keep a quiet socket alive with no spurious reconnect, (c) the watchdog never dials a non-`OPEN` socket. New matrix entry `acp.ws-reconnect-liveness`. Rust unit test `heartbeat_frame_shape_is_stable` pins the wire shape the client keys on.

**TUI / CLI (e2e test)**
- [x] N/A: this PR doesn't change TUI rendering, a CLI subcommand, or session lifecycle

## AI Usage

- [x] AI was used

**AI Model/Tool used:** Claude (via Claude Code, agent-of-empires `/aoe-implement` skill)

**Any Additional AI Details you'd like to share:**
Investigation, plan, and implementation drafted by Claude under my direction, with a `/debate` consult on the threshold and reconnect-vs-close design. I made the design calls and reviewed each commit.

- [x] I am an AI Agent filling out this form (check box if true)

---
<a href="https://app.blacksmith.sh/agent-of-empires/codesmith/agent-of-empires/pr/2324"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-dark-v2.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-light-v2.svg">View with Codesmith</picture></a> <a href="https://backend.blacksmith.sh/track/enable-autofix?expires=1784704689&installation_id=139138837&pr_number=2324&repository=agent-of-empires%2Fagent-of-empires&return_to=https%3A%2F%2Fgithub.com%2Fagent-of-empires%2Fagent-of-empires%2Fpull%2F2324&signature=11e4c584d97c5f41e4713d372eabc600fecdb4893acc1464f72594b31c89e113"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-light.svg">Autofix with Codesmith</picture></a>
<sup>Need help on this PR? Tag <code>/codesmith</code> with what you need. Autofix is disabled.</sup>

## Overview

This PR fixes structured-view WebSocket “silent stalls” behind reverse proxies (e.g., Tailscale Serve, Cloudflare) by adding an application-level heartbeat and a client-side liveness watchdog so dead connections are detected and automatically re-dialed—without requiring user interaction.

## What Changed (high level)

**Added**
- **Server heartbeat frames:** The daemon now sends a small, client-visible `{"kind":"heartbeat"}` text message on its existing keepalive cadence.
- **Client liveness watchdog:** The browser tracks the timestamp of the last message it received (including heartbeats). If an `OPEN` socket goes quiet too long, the client treats it as dead and forces a reconnect.
- **Safer reconnect behavior:** The reconnect logic was adjusted so “old socket close events” can’t accidentally interfere with the newest connection attempt.
- **Visibility/online recovery support:** The client will also re-dial when the tab comes back into view or the browser regains online status (to recover from stale `OPEN` sockets).
- **Documentation:** Troubleshooting guidance was expanded to explain how half-open WebSockets can get stuck behind proxies and how the new recovery works.
- **Tests + coverage:** New/updated unit tests validate the exact heartbeat shape and the watchdog reconnect behavior across important edge cases, plus an added coverage-matrix entry.

**Modified**
- **Reconnect safety in the hook:** The hook’s internal reconnect “dial generation” ordering and retry reset behavior were updated to make reconnection deterministic and resilient to proxy/socket quirks.

## Benefits

- **No more frozen structured views:** The UI recovers automatically after proxy idle timeouts.
- **No manual “poke” required:** Users no longer need to send a message to trigger recovery.
- **No data loss:** Reconnect resumes from the last known sequence and deduplicates correctly, so stale replay can’t cause duplicates or missing updates.
- **Works across proxies:** Designed to handle environments where proxies abruptly reset idle WebSockets without properly triggering a browser-visible close.

## Potential areas for further enhancement
- Add observability hooks (e.g., metrics/logging for watchdog-triggered reconnects) so operators can measure how often “zombie” stalls occur and confirm effectiveness in production.

### CODE REVIEW

File: docs/structured-view/troubleshooting.md [hunk 1]
```diff
@@ -302,6 +302,15 @@ shows `Reconnecting (N/7) in Xs...` while the auto-retry is armed, and a manual
...
 foreground triggers an immediate reconnect.
 
+Behind a reverse proxy (Tailscale Serve, Cloudflare, nginx) an idle connection
+can be reset without the browser ever seeing a close, leaving the socket in a
+half-open state the browser still reports as connected. To catch this, the
+daemon emits a small heartbeat frame every 30s and the view runs a staleness
+watchdog: if no frame (event or heartbeat) arrives for 75s, it treats the
+socket as dead and re-dials, even while the tab is foregrounded and idle. So a
+session left open behind a proxy resyncs on its own instead of silently
+stalling until you send a message.
```
Comment (njbrake) [inline docs/structured-view/troubleshooting.md]: @Seluj78 comments like this feel uneccessary for docs.

Comment (njbrake) [inline docs/structured-view/troubleshooting.md]:
```suggestion
```

**Review by njbrake — APPROVED** [review]
